src/ceph_mon.cc

```diff
@@ -614,28 +614,42 @@ int main(int argc, const char **argv)
   } else {
     dout(0) << g_conf->name << " does not exist in monmap, will attempt to join an existing cluster" << dendl;
 
-    pick_addresses(g_ceph_context, CEPH_PICK_ADDRESS_PUBLIC);
-    if (!g_conf->public_addr.is_blank_ip()) {
-      ipaddr = g_conf->public_addr;
-      if (ipaddr.get_port() == 0)
-	ipaddr.set_port(CEPH_MON_PORT);
-      dout(0) << "using public_addr " << g_conf->public_addr << " -> "
-	      << ipaddr << dendl;
-    } else {
-      MonMap tmpmap;
-      int err = tmpmap.build_initial(g_ceph_context, cerr);
-      if (err < 0) {
-	derr << argv[0] << ": error generating initial monmap: "
-             << cpp_strerror(err) << dendl;
-	usage();
-	prefork.exit(1);
+    entity_addr_t conf_host;
+    std::vector <std::string> my_sections;
+    g_conf->get_my_sections(my_sections);
+    std::string mon_host_str;
+    if (g_conf->get_val_from_conf_file(my_sections, "mon host",
```

Comment:
Just update the code with a new approach. As per the commit message says, it does add all the possible IPs to try into a list, add the public network to it and then try to bind against all IPs in the list before making a fault.